OpenRA.Game/Game.cs

```diff
@@ -855,6 +855,12 @@ static RunStatus Run()
 			return state;
 		}
 
+		public static void ExitIfNotMidGame()
+		{
+			if (OrderManager.Connection.ConnectionState != ConnectionState.Connected)
```

Comment:
The problem with this approach is that Cmd+q will quit if you're in e.g. the multiplayer server browser, but not if you're in a skirmish lobby. This distinction feels pretty arbitrary, and it wouldn't be fair to scope creep this PR to require more advanced / intuitive logic. So IMO it would be better to just universally block it for now.